site/js/modules/webui/course.test.js

```diff
@@ -0,0 +1,56 @@
+import * as Base from './base.js';
+import * as Event from './event.js';
+import * as Routing from './routing.js';
+import * as TestUtil from './test/util.js';
+
+test("Enrolled Courses", async function() {
+    Base.init(false);
+
+    await TestUtil.loginUser("course-student");
+    await navigateToEnrolledCourses();
+
+    expect(document.title).toContain('Enrolled Courses');
+
+    let pageContent = document.querySelector('.page-body .content[data-page="enrolled courses"]');
+    expect(pageContent).not.toBeNull();
+
+    const courseCardSpans = pageContent.querySelectorAll('.cards-area .card span');
+    expect(courseCardSpans[0].textContent).toBe('Course 101');
+    expect(courseCardSpans[1].textContent).toBe('Course Using Different Languages');
+});
+
+test("Nav Course101", async function() {
+    Base.init(false);
+
+    await TestUtil.loginUser("course-student");
+
+    let targetCourse = 'course101';
+    await navigateToCourse(targetCourse);
+
+    expect(document.title).toContain(targetCourse);
+
+    let pageContent = document.querySelector('.page-body .content[data-page="course"]');
+    expect(pageContent.querySelector('h2').textContent).toBe('Course 101');
+    expect(pageContent.querySelector(`.cards-area .card-assignment a[href="#course/assignment?assignment=hw0&course=${targetCourse}"]`)).not.toBeNull();
+});
+
+async function navigateToEnrolledCourses() {
+    let coursesRenderedPromise = Event.getEventPromise(Event.EVENT_TYPE_HANDLER_COMPLETED, {
+        'path': 'courses',
+    });
+
+    Routing.route('#courses');
+    await coursesRenderedPromise;
+}
+
+async function navigateToCourse(courseID) {
+    let courseRenderedPromise = Event.getEventPromise(Event.EVENT_TYPE_HANDLER_COMPLETED, {
+        'path': 'course',
+        'params': {
+            'course': 'course101',
+        },
+    });
+
+    Routing.route('#course?course=course101');
```

Comment:
I am a little wary here of using the hardcoded paths,
since we already establish them as constants in the routing file.

I think the solution here would be to make some routing function (routeComponents()?)
that accepts the routing path and params.
Then we can use the constants.
There also may be a trick here you can use in the event system.
Like if you had `function routeComponents({path: undefined, params: {}})` or something like that you may be able to pass the same thing to both the event system and routing.